src/libraries/Common/tests/SourceGenerators/RoslynTestUtils.cs

```diff
@@ -0,0 +1,296 @@
+        }
+
+        public static Task CommitChanges(this Project proj, params string[] ignorables)
+        {
+            Assert.True(proj.Solution.Workspace.TryApplyChanges(proj.Solution));
+            return AssertNoDiagnostic(proj, ignorables);
+        }
+
+        public static async Task AssertNoDiagnostic(this Project proj, params string[] ignorables)
+        {
+            foreach (Document doc in proj.Documents)
+            {
+                SemanticModel sm = await doc.GetSemanticModelAsync(CancellationToken.None).ConfigureAwait(false);
+                Assert.NotNull(sm);
+
+                foreach (Diagnostic d in sm!.GetDiagnostics())
+                {
+                    bool ignore = ignorables.Any(ig => d.Id == ig);
+
+                    Assert.True(ignore, d.ToString());
+                }
+            }
+        }
+
+        public static Project WithDocument(this Project proj, string name, string text)
+        {
+            return proj.AddDocument(name, text).Project;
+        }
+
+        public static Document FindDocument(this Project proj, string name)
+        {
+            foreach (Document doc in proj.Documents)
+            {
+                if (doc.Name == name)
+                {
+                    return doc;
+                }
+            }
+
+            throw new FileNotFoundException(name);
+        }
+
+        /// <summary>
+        /// Looks for /*N+*/ and /*-N*/ markers in a string and creates a TextSpan containing the enclosed text.
+        /// </summary>
+        public static TextSpan MakeSpan(string text, int spanNum)
+        {
+            int start = text.IndexOf($"/*{spanNum}+*/", StringComparison.Ordinal);
+            if (start < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(spanNum));
+            }
+
+            start += 6;
+
+            int end = text.IndexOf($"/*-{spanNum}*/", StringComparison.Ordinal);
+            if (end < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(spanNum));
+            }
+
+            end -= 1;
+
+            return new TextSpan(start, end - start);
+        }
+
+        /// <summary>
+        /// Runs a Roslyn generator over a set of source files.
+        /// </summary>
+        public static async Task<(ImmutableArray<Diagnostic>, ImmutableArray<GeneratedSourceResult>)> RunGenerator(
+            ISourceGenerator generator,
+            IEnumerable<Assembly>? references,
+            IEnumerable<string> sources,
+            AnalyzerConfigOptionsProvider? optionsProvider = null,
+            bool includeBaseReferences = true,
+            CancellationToken cancellationToken = default)
+        {
+            Project proj = CreateTestProject(references, includeBaseReferences);
+
+            int count = 0;
+            foreach (string s in sources)
+            {
+                proj = proj.WithDocument($"src-{count++}.cs", s);
```

Comment:
Seems like this is duplicated around multiple places. Maybe add an extension method `WithDocuments` that does the for loop and adds all the documents with name `src-n.cs`?